omero_marshal/encode/encoders/shape.py

```diff
@@ -33,7 +35,53 @@ def encode(self, obj):
         self.set_if_not_none(v, 'TheC', obj.theC)
         self.set_if_not_none(v, 'TheT', obj.theT)
         self.set_if_not_none(v, 'TheZ', obj.theZ)
+        self.set_if_not_none(
+            v, 'Transform', self.encode_transform(obj.transform))
         self.set_if_not_none(v, 'Visible', obj.visibility)
         return v
 
+    @staticmethod
+    def encode_transform(transform):
+        transform = unwrap(transform)
+        if not transform or transform == 'none':
+            return
+
+        tr, args = transform[:-1].split('(')
+        a = map(float, args.split(' '))
+
+        if tr == 'matrix':
+            pass
+        elif tr == 'translate':
+            a = [1.0, 0.0, 0.0, 1.0, a[0], a[1] if len(a) > 1 else 0.0]
+        elif tr == 'scale':
+            a = [a[0], 0.0, 0.0, a[-1], 0.0, 0.0]
+        elif tr == 'rotate':
```

Comment:
The composition performed here may be useful to @mtbc in case we want to avoid implementing or deal with matrix dot products in SQL. :) I know that work has already been done and merged but it could simplify the PL/SQL a great deal.

/cc @joshmoore, @jburel, @sbesson